__tests__/app.test.js

```diff
@@ -233,5 +234,93 @@ afterAll(() => {
         })
     }) 
     
-
+    describe("POST /api/articles/:article_id/comments", () => {
+        test("201: should add a new comment consisting of a username and a body for the specified article id parameter", () => {
+            const newComment = {
+                body: "hello im a comment",
+                username: "lurker"
+            }
+            return request(app)
+                .post("/api/articles/2/comments")
+                .send(newComment)
+                .expect(201)
+                .then(({body}) => {
+                    expect(body.comment).toMatchObject({
+                        comment_id: 19,
+                        body: "hello im a comment",
+                        article_id: 2,
+                        author: "lurker",
+                        votes: expect.any(Number),
+                        created_at: expect.any(String)
+                    })
+            })
+        })
+        test("201: should add a new comment consisting of a username and a body, and ignore any other info passed", () => {
+            const newComment = {
+                body: "hello im a comment",
+                username: "lurker",
+                email: "lurks@gmail.com"
+            }
+            return request(app)
+                .post("/api/articles/2/comments")
+                .send(newComment)
+                .expect(201)
+                .then(({body}) => {
+                    expect(body.comment).toMatchObject({
+                        comment_id: 19,
+                        body: "hello im a comment",
+                        article_id: 2,
+                        author: "lurker",
+                        votes: expect.any(Number),
+                        created_at: expect.any(String)
+                    })
+            })
+        })
+        test("400: should return a 400 malformed request error when request body is missing", () => {
+            const newComment = {
+                username: "lurker",
+            }
+            return request(app)
+                .post("/api/articles/2/comments")
+                .send(newComment)
+                .expect(400)
+                .then(({body}) =>{
+                    expect(body.msg).toBe("Bad request")
+                })
+        })
+        test("400: should return a 400 malformed request error when username isn't valid", () => {
```

Comment:
if username wasn't valid it would be a 404

Reply:
hi, can you clarify please as i was thinking 400 because it's trying to access the username based on the users database table...